### PR TITLE
Add native draft status to Space tasks

### PR DESCRIPTION
**Pull request by lsm**
## Summary

- Add first-class `draft` status to `SpaceTaskStatus` so tasks can be created as drafts that the orchestrator never auto-starts
- Draft → open transition only via explicit publish action; all other transitions out of draft are rejected
- Adds `publishTask` manager method, `spaceTask.publish` RPC, `draft` param on `create_standalone_task` MCP tool
- Migration 114 updates `space_tasks` CHECK constraint to include `draft`
- Frontend: transition labels, badge colors, filters, and SpaceStore publish method

## Test plan

- [x] 24 new unit tests covering draft lifecycle (creation, transitions, publish, dependency blocking)
- [x] 3 new RPC handler tests for `spaceTask.publish`
- [x] All existing tests pass (2376 daemon unit tests, 148 web tests)
- [x] Type check, lint, format all pass
- [ ] Manual: create draft task via UI, verify not auto-started, publish to open

Closes #224

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Opus 4.7 (Anthropic)

> **Model:** claude-opus-4-7 | **Client:** NeoKai | **Provider:** Anthropic

### Verdict: REQUEST CHANGES

Overall this is a solid, well-structured implementation. The core data model (status type, transitions, migration, manager method) is correct and well-tested. The orchestrator is safely guarded by virtue of filtering `status === 'open'` in its tick loops — draft tasks (`status === 'draft'`) are never picked up. Migration 114 uses a safe dynamic DDL approach with proper FK-off/try-finally. All 39 CI checks pass.

However, there are two P1 issues and several P2 items to address before this ships.

---

### P1 — Should-fix

**1. `list_tasks` MCP tool missing `draft` in status enum**

`packages/daemon/src/lib/space/tools/space-agent-tools.ts` line 1037

```ts
z.enum(['open', 'in_progress', 'done', 'blocked', 'cancelled', 'archived'])
```

Agents can now create draft tasks via `create_standalone_task({ draft: true })`, but the `list_tasks` tool's Zod schema rejects `'draft'` at parse time. An agent that creates a draft task has no way to query for it later. `'review'` and `'approved'` are also missing (pre-existing) — please add all three.

**2. `draft: true` + `status: 'open'` silently creates a non-draft task**

`packages/daemon/src/lib/rpc-handlers/space-task-handlers.ts` lines 63–66

```ts
if (draft && !rest.status) {
    rest.status = 'draft';
}
```

If a caller sends both `draft: true` and `status: 'open'`, the `status: 'open'` wins and the task is created as open — the `draft` flag is silently ignored. This is surprising. Either:
- (a) Make `draft: true` override `status` unconditionally (i.e., remove the `!rest.status` guard), or
- (b) Throw if both `draft: true` and `status` are provided and `status !== 'draft'`.

Option (b) is safer — it prevents ambiguous API usage.

---

### P2 — Suggestions

**3. UI draft→open transition bypasses dedicated `publishTask` RPC**

`packages/web/src/components/space/SpaceTaskPane.tsx` line 588

The `handleStatusTransition` function routes `draft → open` through the generic `spaceStore.updateTask(task.id, { status: 'open' })`. This works (the transition map enforces it), but it bypasses the `spaceTask.publish` RPC's explicit draft-status validation and clearer error messages. Since a `publishTask` method was added to `spaceStore` (line 1594), the UI should use it:

```ts
if (task.status === 'draft' && newStatus === 'open') {
    await spaceStore.publishTask(task.id);
} else {
    await spaceStore.updateTask(task.id, { status: newStatus });
}
```

**4. `spaceStore.publishTask` is dead code**

`packages/web/src/lib/space-store.ts` lines 1594–1605

The `publishTask` method was added to the store but is never called from any component. Either wire it up (see P2 #3) or remove it.

**5. Missing edge-case test coverage**

`packages/daemon/tests/unit/5-space/runtime/task-draft-status.test.ts`

The test suite covers the core lifecycle well, but a few important combinations are missing:
- Creating a draft task with `depends_on` referencing another draft task (both in draft) — should dependency creation succeed?
- Creating a task with `draft: true` AND `workflow_id` set — verify it does NOT auto-attach to a workflow
- Calling `publishTask` on a non-draft task via the RPC handler — test verifies the error message

---

### P3 — Nits

**6. Pre-existing: `promoteDraftTasksByCreator` SQL matches wrong status**

`packages/daemon/src/storage/repositories/space-task-repository.ts` line 367

```sql
WHERE created_by_task_id = ? AND status = 'open'
```

This sets `status = 'open'` where `status = 'open'` — a no-op. Now that `draft` is a first-class status, this should match `status = 'draft'`. Currently dead code in Space context, but will bite when wired up for planning workflows. Not introduced by this PR, but worth fixing while you're in the area.

File: packages/web/src/lib/task-filters.ts [hunk 1]
```diff
@@ -49,7 +49,20 @@ export function isActionRequired(task: ActionRequiredTaskInput): boolean {
  *   `postApprovalBlockedReason` populated because dispatch failed)
  *   visible to the user. The task-detail pane's
  *   `PendingPostApprovalBanner` surfaces the actionable failure.
+ *
+ * Excludes:
+ * - `draft`: draft tasks are not yet runnable. They appear in a separate
+ *   "Drafts" grouping.
  */
 export function isActiveTask(task: ActionRequiredTaskInput): boolean {
 	return task.status === 'open' || task.status === 'in_progress' || task.status === 'approved';
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/web/src/lib/task-filters.ts:58]: Addressed in this branch: draft tasks now have a dedicated Drafts tab/type route so they are included in the task list navigation and can be opened/published from the UI instead of being filtered out of every tab.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Opus 4.7 (Anthropic)

> **Model:** claude-opus-4-7 | **Client:** NeoKai | **Provider:** Anthropic

### Verdict: REQUEST CHANGES

This is a well-executed feature across all layers — type definition, migration, manager methods, RPC handlers, MCP tools, and frontend. The orchestrator is structurally safe: it filters `status === 'open'` at three independent checkpoints, so draft tasks are never auto-started. The migration uses a battle-tested dynamic DDL pattern with FK-off/try-finally and an idempotency guard. Previous review findings (list_tasks enum, draft+status contradiction, Drafts tab) are all addressed.

One P1 and two P2s remain.

---

### P1 — Should-fix

**Stale test for `promoteDraftTasksByCreator` does not verify the fix**

`packages/daemon/tests/unit/4-space-storage/storage/space-task-repository.test.ts` lines 357–383

The PR correctly fixes a pre-existing bug: the WHERE clause changed from `status = 'open'` (no-op) to `status = 'draft'` (actually promotes drafts). However, the existing test was not updated to exercise the new behavior:

- It creates tasks with `status: 'open'` and `status: 'in_progress'` — neither matches the new `WHERE status = 'draft'` clause
- The assertion `expect(count).toBeGreaterThanOrEqual(0)` trivially passes with 0 changes
- It does NOT create a `draft` task and verify it gets promoted to `open`

The test should be updated to:
1. Create a task with `status: 'draft'` and `createdByTaskId: 'planner-1'`
2. Call `promoteDraftTasksByCreator('planner-1')`
3. Assert `count === 1` and the task is now `open`

Also update the stale comment on line 359: `"promoteDraftTasksByCreator targets open tasks (was 'draft' before M71 → now 'open')"` — this is now misleading since it targets `draft` again.

---

### P2 — Suggestions

**1. Stale docstring on `promoteDraftTasksByCreator`**

`packages/daemon/src/storage/repositories/space-task-repository.ts` line 362:

```ts
/** Promote open tasks created by a planning task (legacy method, kept for API compatibility) */
```

The function now promotes `draft` tasks → `open`. Update the docstring to reflect the actual behavior.

**2. `includeDrafts` API parameter not implemented (acceptance criteria gap)**

The task description says: _"`GET /tasks` and `list_tasks` should filter draft tasks out of the default 'active' view, with an opt-in `includeDrafts` flag"_. The UI correctly filters drafts into a separate tab via predicates, but the backend API (`listBySpace`) and the MCP `list_tasks` tool return all non-archived tasks including drafts unconditionally. An agent calling `list_tasks` without a status filter will see draft tasks mixed in with actionable ones. This is a minor UX concern for agent consumers but worth noting as a follow-up.

---

### What looks good

- **Orchestrator enforcement**: Three independent `status === 'open'` guards in `attachStandaloneTasksToWorkflows` — structurally airtight
- **Transition table**: `draft: ['open', 'archived']` — minimal, correct
- **Migration 114**: Dynamic DDL derivation, idempotency guard (`statusCheckContains`), FK-off/try-finally — follows existing patterns (M104)
- **Draft+status contradiction**: Handler rejects `draft: true` combined with a non-draft status
- **Draft tab UI**: Conditionally shown when drafts exist, clean integration with existing tab system
- **Dependency handling**: `areDependenciesMet` correctly returns `false` for draft deps; well-tested
- **Test coverage**: 315 new lines of draft lifecycle tests covering transitions, dependencies, publish flow, and edge cases

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Opus 4.7 (Anthropic)

> **Model:** claude-opus-4-7 | **Client:** NeoKai | **Provider:** Anthropic

### Verdict: APPROVE ✅

All findings from the previous review round have been addressed:

- **P1 stale test** (`space-task-repository.test.ts`): Now creates a `draft` task, asserts `count === 1`, and verifies the promoted task is `open` while the `in_progress` task stays unchanged. The stale comment is also fixed.
- **P2 stale docstring** (`space-task-repository.ts`): Updated "Promote open tasks" → "Promote draft tasks".
- **P2 `includeDrafts` API param**: Acknowledged as follow-up — reasonable scope decision since the UI filtering is correct and the backend returns drafts for the overview data path.

Bonus: `space-test-db.ts` helper updated to include `draft` in the CHECK constraint so the test exercises the real schema path.

Clean, focused changeset. No new findings. Ship it. 🚀
